### PR TITLE
Use the INCOMING_HOOK_BODY env variable.

### PR DESCRIPTION
**Pull request by willrlin**
## Issue

Addresses #58 
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This variable is used to override BACKEND_API_BASE for any logic that is
sent to the client; i.e. anything in ui-components/. This allows us to
specify different URLs for the logic running in the client and the logic
running inside the container.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Run `docker-compose up` and attempt to login via the public site.
- [x] Desktop View
- [x] Mobile View

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,9 +17,11 @@ services:
     env_file:
       - ./sites/public/.env
     environment:
-      # TODO(CityOfDetroit/affordable-housing-app#84): login currently doesn't work, likely
-      # needs a separate env variable to point client requests to localhost:3100 instead.
+      # The URLs are different here because requests made using BACKEND_API_BASE are done from
+      # the NextJS server, which must address other containers by container name. Requests made
+      # using INCOMING_HOOK_BODY are done from the client side browser and must use localhost.
       BACKEND_API_BASE: "http://backend-core:3100"
+      INCOMING_HOOK_BODY: "http://localhost:3100"
```
Comment (anders-schneider) [inline docker-compose.yml:24]: I'm skimming https://docs.netlify.com/configure-builds/environment-variables/#build-hook-metadata-and-payload and https://docs.netlify.com/configure-builds/build-hooks/ to try to understand what this is doing: there's a POST request happening somewhere, and the body of that POST is now "http://localhost:3100". Is there anything on the other end, receiving the POST request, that knows what to do with that message body? (i.e. somewhere where we say "whatever was in the body of the request, swap that in for BACKEND_API_BASE"?)

**Review by anders-schneider — APPROVED** [review]
I understand maybe 50% of what this change is doing, and recorded what I do/don't understand in a question. I'm approving, though, in the interest of fast progress :)

File: docker-compose.yml (hunk 1 above)
Comment (willrlin) [reply docker-compose.yml:24]: This change isn't really related to the Netlify variable itself or what that does when deploying it on Netlify; just the fact that it's being used here https://github.com/CityOfDetroit/bloom/blob/main/sites/public/next.config.js to override the BACKEND_API_BASE only for ui-components. Which, I think is needed when deploying this for the exact reason that we need it here (client side logic needing a different API address).

**Comment by willrlin** [comment]
I don't think you need to attempt to understand the Netlify variable itself, just that it's being used to override the backend address that the client code talks to (as opposed to the NextJS frontend server).

File: docker-compose.yml (hunk 1 above)
Comment (anders-schneider) [reply docker-compose.yml:24]: Grrr, Github search failed me: https://screenshot.googleplex.com/4HgfQyQVcy8aLpG

If I'd've seen that we were consuming INCOMING_HOOK_BODY somewhere, I wouldn't have gone off on my wild goose chase to figure out what that variable was. Oh well, thanks for pointing to `next.config.js`!

(and yes, this is enough evidence that I should just use VSCode to search 🙃 )

**Pull request merged**